### DISCUSSION TITLE
Error in NFT Project

### DISCUSSION THREAD

**Discussion by JDEEPSingh04** [Cyfrin/foundry-full-course-cu#1037] [Q&A]
I am working on the NFT project
Here are my files:
BasicNFT.sol
``Solidity
// SPDX-License-Identifier: MIT 
pragma solidity ^0.8.0;

import {ERC721} from "lib/openzeppelin-contracts/contracts/token/ERC721/ERC721.sol";

contract BasicNFT is ERC721{
    uint256 s_TokenCounter=0;
    mapping(uint256=>string) private s_TokenIDtoURI;

    constructor()ERC721("Doggie","Dog"){
        s_TokenCounter=0;
    }
    function mintNFT(string memory TokenURI) public{
        s_TokenIDtoURI[s_TokenCounter]=TokenURI;
        _safeMint(msg.sender,s_TokenCounter);
        s_TokenCounter++;
    }
    function tokenURI(uint256 TokenID) public view override returns(string memory){
        return s_TokenIDtoURI[TokenID];
    }
}
``

Interactions.sol

// SPDX-License-Identifier: MIT
pragma solidity ^0.8.0;

import {BasicNFT} from "../src/BasicNFT.sol";
import {DeployBasicNFT} from "../script/DeployBasicNFT.s.sol";
import {Script} from "forge-std/Script.sol";
import {DevOpsTools} from "lib/foundry-devops/src/DevOpsTools.sol";

contract MintBasicNFT is Script {
    string public constant URI =
        "ipfs://bafybeig37ioir76s7mg5oobetncojcm3c3hxasyd4rvid4jqhy4gkaheg4/?filename=0-PUG.json";

    function run() external {
        address MostRecentDeployed = DevOpsTools.get_most_recent_deployment(
            "BasicNFT",
            block.chainid
        );
        mintNFTonContract(MostRecentDeployed);
    }

    function mintNFTonContract(address MostRecentDeployed) public {
        vm.startBroadcast();
        BasicNFT(0x3e24F742fF1329de8442a9B0D198A4a4d2BcF986).mintNFT(URI);
        vm.stopBroadcast();
    }
}

DeployScript
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.0;

import {Script} from "forge-std/Script.sol";
import {BasicNFT} from "../src/BasicNFT.sol";

contract DeployBasicNFT is Script{
    function run() external returns(BasicNFT){
        vm.startBroadcast();
        BasicNFT nft=new BasicNFT();
        vm.stopBroadcast();
        return nft;
    }
}

When I deploy the contract, it gets deployed on sepolia testnet
But after that when i try to call the mintNFT function it gives an error contract not deployed
ERROR apply:ext: evm::cheatcodes: non-empty stderr args=["bash", "/Users/jagdeep/Desktop/foundry/foundary-f23/foundry-NFT-f23/lib/foundry-devops/src/get_recent_deployment.sh", "BasicNFT", "11155111", "/Users/jagdeep/Desktop/foundry/foundary-f23/foundry-NFT-f23//broadcast"] stderr="/Users/jagdeep/Desktop/foundry/foundary-f23/foundry-NFT-f23/lib/foundry-devops/src/get_recent_deployment.sh: line 31: jq: command not found\n/Users/jagdeep/Desktop/foundry/foundary-f23/foundry-NFT-f23/lib/foundry-devops/src/get_recent_deployment.sh: line 33: jq: command not found\n/Users/jagdeep/Desktop/foundry/foundary-f23/foundry-NFT-f23/lib/foundry-devops/src/get_recent_deployment.sh: line 35: ((: i<: syntax error: operand expected (error token is \"<\")\n"

**Comment by GenevaRoc** [comment]
Did you install makefile yet?

**Reply by JDEEPSingh04** [reply]
Yes I have installed makefile

**Comment by GenevaRoc** [comment]
Did you do ffi = true?

**Reply by JDEEPSingh04** [reply]
hmm

**Comment by GenevaRoc** [comment]
Ah-hah, try installing “Sudo apt install jp”

**Reply by JDEEPSingh04** [reply]
Okk let me try

**Reply by JDEEPSingh04** [reply]
Thanks it worked with this solution
The error was jp not installed
